### PR TITLE
Add UDP statistics

### PR DESCRIPTION
**Pull request by bwester**
Collect and report basic UDP statistics along with TCP statistics. This
implementation generalizes the TCP-handling code, since the TCP and UDP
stats come from the same file.

### CODE REVIEW

File: os/netstat/netstat_linux.go [hunk 1]
```diff
@@ -31,9 +41,16 @@ type TCPStat struct {
...
+}
+
+// UDPStat represents statistics about UDP. Reflection is used to map field names to
+// kernel data.
+type UDPStat struct {
+	InDatagrams  *metrics.Counter
+	InErrors     *metrics.Counter
+	OutDatagrams *metrics.Counter
+	RcvbufErrors *metrics.Counter
+	SndbufErrors *metrics.Counter
```
Comment (nerdrew) [inline os/netstat/netstat_linux.go]: Do we want to monitor `NoPorts` errors too?

Comment (nerdrew) [inline os/netstat/netstat_linux.go]: Based on reading this blog: https://blog.packagecloud.io/eng/2016/06/22/monitoring-tuning-linux-networking-stack-receiving-data/#procnetsnmp

**Comment by nerdrew** [comment]
LGTM

File: os/netstat/netstat_linux.go (hunk 1 above)
Comment (bwester) [inline os/netstat/netstat_linux.go]: Might as well

**Pull request merged**